src/naive/date.rs

```diff
@@ -1885,7 +1733,7 @@ impl Iterator for NaiveDateDaysIterator {
         // current < NaiveDate::MAX from here on:
         let current = self.value;
         // This can't panic because current is < NaiveDate::MAX:
-        self.value = current.succ();
+        self.value = current.succ_opt().unwrap();
```

Comment:
could we use `self.value = current.succ_opt()?;` and then remove the whole `if` block?

Reply:
I played with something like that at the time, but I don't think so? In the case `succ_opt()` returns `None` but current is `Some` we still have to return that last value. Arguably the iterator contract should be changed but I didn't want to take that on as part of this PR.

Reply:
sounds good - I just remembered why this stood out to me, I've made some changes to it in https://github.com/chronotope/chrono/pull/824/commits/c7e1b9b8deee5cb378766d32d16cabed7184125a anyway, so lets merge this as-is